libcontainer/container_linux.go

```diff
@@ -2246,6 +2250,34 @@ func (c *Container) bootstrapData(cloneFlags uintptr, nsMaps map[configs.Namespa
 		})
 	}
 
+	// write boottime and monotonic offsets
+	// 0 is default value in /proc/PID/timens_offsets
+	// if 0, do not bootstrap data
+	if c.config.TimeOffsets != nil {
+
+		if c.config.TimeOffsets["boottime"].Nanosecs != 0 || c.config.TimeOffsets["boottime"].Secs != 0 {
+			b, err := encodeTimeNs(c.config.TimeOffsets["boottime"])
+			if err != nil {
+				return nil, err
+			}
+			r.AddData(&Bytemsg{
+				Type:  BootTimeNsAttr,
+				Value: append([]byte("boottime "), b...),
+			})
+		}
+
+		if c.config.TimeOffsets["monotonic"].Nanosecs != 0 || c.config.TimeOffsets["monotonic"].Secs != 0 {
+			b, err := encodeTimeNs(c.config.TimeOffsets["monotonic"])
+			if err != nil {
+				return nil, err
+			}
+			r.AddData(&Bytemsg{
+				Type:  MonotonicNsAttr,
+				Value: append([]byte("monotonic "), b...),
+			})
+		}
+	}
+
```

Comment:
These can be written with one string by separating them with newlines, so there's no need to add multiple netlink entries for this. Just generate one string into a `TimeOffsetsAttr` and write it in one shot in `nsexec.c`.